idea-plugin/src/main/kotlin/arrow/meta/ide/plugins/quotes/lifecycle/QuoteLifecycle.kt

```diff
@@ -0,0 +1,189 @@
+import arrow.meta.ide.dsl.utils.quoteRelevantFiles
+import arrow.meta.ide.phases.resolve.LOG
+import arrow.meta.ide.plugins.quotes.cache.QuoteCache
+import arrow.meta.ide.plugins.quotes.resolve.QuoteHighlightingCache
+import arrow.meta.ide.plugins.quotes.system.QuoteSystemService
+import arrow.meta.ide.plugins.quotes.system.cacheStrategy
+import arrow.meta.ide.testing.UnavailableServices
+import arrow.meta.ide.testing.unavailableServices
+import arrow.meta.quotes.analysisIdeExtensions
+import com.intellij.openapi.application.Application
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.application.ModalityState
+import com.intellij.openapi.application.ReadAction
+import com.intellij.openapi.application.runReadAction
+import com.intellij.openapi.editor.Document
+import com.intellij.openapi.editor.EditorFactory
+import com.intellij.openapi.editor.event.BulkAwareDocumentListener
+import com.intellij.openapi.fileEditor.FileDocumentManager
+import com.intellij.openapi.progress.EmptyProgressIndicator
+import com.intellij.openapi.progress.ProgressIndicator
+import com.intellij.openapi.progress.ProgressManager
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.roots.FileIndexFacade
+import com.intellij.openapi.util.Key
+import com.intellij.openapi.vfs.AsyncFileListener
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.openapi.vfs.VirtualFileManager
+import com.intellij.openapi.vfs.newvfs.events.VFileContentChangeEvent
+import com.intellij.openapi.vfs.newvfs.events.VFileCopyEvent
+import com.intellij.openapi.vfs.newvfs.events.VFileEvent
+import com.intellij.openapi.vfs.newvfs.events.VFileMoveEvent
+import com.intellij.psi.PsiDocumentManager
+import com.intellij.testFramework.LightVirtualFile
+import com.intellij.util.concurrency.BoundedTaskExecutor
+import com.intellij.util.ui.update.Update
+import org.jetbrains.kotlin.idea.debugger.readAction
+import org.jetbrains.kotlin.psi.KtFile
+import org.jetbrains.kotlin.utils.addToStdlib.safeAs
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicBoolean
+
+data class QuoteConfigs(
+  val quoteSystem: QuoteSystemService,
+  val cache: QuoteCache
+)
+
+fun Project.quoteConfigs(): QuoteConfigs? =
+  getService(QuoteSystemService::class.java)?.let { quoteSystem ->
+    getService(QuoteCache::class.java)?.let { cache ->
+      QuoteConfigs(quoteSystem, cache)
+    }
+  } ?: throw UnavailableServices(listOf(QuoteSystemService::class.java, QuoteCache::class.java))
+
+internal fun quoteProjectOpened(project: Project): Unit = // previously in QuoteSystemCache.projectOpend()
+  project.quoteConfigs()?.let { (quoteSystem: QuoteSystemService, cache: QuoteCache) ->
+    // add a startup activity to populate the cache with a transformation of all project files
+    //StartupManager.getInstance(project).runWhenProjectIsInitialized {
+    //runBackgroundableTask("Arrow Meta", project, cancellable = false) {
```

Comment:
Removing the task here is a change of behaviour. 
`quoteProjectOpened` is executed on the EDT, so any long-running tasks shouldn't be performed in this thread. Otherwise the UI would block. 
It's executed on the EDT, because it's wrapped in a non-dumbaware startup activity, see the [docs of StartupActivity](https://github.com/JetBrains/intellij-community/blob/193/platform/core-api/src/com/intellij/openapi/startup/StartupActivity.java#L11) for the threading guidelines.

imo at least the use of the task should be re-activated here.

Reply:
made the Startup activity dumbAware in the last commit